irods/manager/user_manager.py

```diff
@@ -82,6 +82,76 @@ def temp_password_for_user(self, user_name):
             return obf.create_temp_password(msg.stringToHashWith, conn.account.password)
 
 
+    class EnvStoredPasswordNotEdited(RuntimeError):
+
+        """
+        Error thrown by a password change attempt if a login password encoded in the
+        irods environment could not be updated.
+
+        This error will be seen when `modify_secrets_file' is set True and the
+        authentication scheme in effect for the session is other than iRODS native,
+        using a password loaded from the client environment.
+	"""
+
+        def __init__(self,msg=None):
+            """Initialize  the exception object (with a message if desired)."""
+            super(UserManager.EnvStoredPasswordNotEdited,self).__init__(msg)
+
+    @staticmethod
+    def abspath_exists(path):
+        return os.path.isabs(path) and os.path.exists(path)
+
+    def modify_own_password(self, old_value, new_value, modify_secrets_file = False):
```

Comment:
so... we agreed to make it `change_password`?    i could have missed some back and forth.

https://github.com/irods/python-irodsclient/pull/329#discussion_r783203017

Reply:
I'm good with any of 
  - `change_own_password`
  - `change_password`
  - `modify_own_password`
  - `modify_password`

Whatever shines best and conveys that this is something different than the iadmin operation of : 

**moduser** user **password** pass

Reply:
done